src/listener3d.js

```diff
@@ -0,0 +1,226 @@
+'use strict'
+
+define(function (require) {
+  var p5sound = require('master');
+  var Effect = require('effect');
+
+  /**
+   * listener is a class that can construct both a Spatial Panner
+   * and a Spatial Listener. The panner is based on the 
+   * Web Audio Spatial Panner Node
+   * https://www.w3.org/TR/webaudio/#the-listenernode-interface
+   * This panner is a spatial processing node that allows audio to be positioned
+   * and oriented in 3D space. 
+   *
+   * The Listener modifies the properties of the Audio Context Listener. 
+   * Both objects types use the same methods. The default is a spatial panner.
+   *
+   * <code>p5.Panner3D</code> - Constructs a Spatial Panner<br/>
+   * <code>p5.Listener3D</code> - Constructs a Spatial Listener<br/>
+   *
+   * @class listener
+   * @constructor
+   * @return {Object} p5.Listener3D Object
+   *
+   * @param {Web Audio Node} listener Web Audio Spatial Panning Node
+   * @param {AudioParam} listener.panningModel "equal power" or "HRTF"
+   * @param {AudioParam} listener.distanceModel "linear", "inverse", or "exponential"
+   * @param {String} [type] [Specify construction of a spatial panner or listener]
+   */
+	p5.Listener3D = function(type) {
+
+    // if (type==="listener") {
+
+      this.ac = p5sound.audiocontext;
+      this.listener = this.ac.listener;
+	}; 
+
+  /**
+   * Connect an audio sorce
+   * @param  {Object} src Input source
+   */
+  p5.Listener3D.prototype.process = function(src) {
+    src.connect(this.input);
+  }
+  /**
+   * Set the X,Y,Z position of the Panner
+   * @param  {[Number]} xVal
+   * @param  {[Number]} yVal
+   * @param  {[Number]} zVal
+   * @param  {[Number]} time
+   * @return {[Array]}      [Updated x, y, z values as an array]
```

Comment:
square brackets are for optional parameters, like this

`@param  {Number} [time]`